docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp-new.rst

```diff
@@ -0,0 +1,433 @@
+.. _analyzing-data-flow-in-cpp-new:
+
+.. pull-quote:: Note
+
+   The data flow library described here is available from CodeQL 2.13.0 onwards. See :ref:`here <analyzing-data-flow-in-cpp>` for the library available in earlier versions.
+
+Analyzing data flow in C and C++
+================================
+
+You can use data flow analysis to track the flow of potentially malicious or insecure data that can cause vulnerabilities in your codebase.
+
+About data flow
+---------------
+
+Data flow analysis computes the possible values that a variable can hold at various points in a program, determining how those values propagate through the program, and where they are used. In CodeQL, you can model both local data flow and global data flow. For a more general introduction to modeling data flow, see ":ref:`About data flow analysis <about-data-flow-analysis>`."
```

Comment:
> Data flow analysis computes the possible values that a variable can hold at various points in a program

I think this sentence might be setting up a misconception - that data flow is tracking values themselves, like range analysis does, whereas in fact it's about the paths values take or the sources of those values.

Reply:
My suggestion would be something like:
> Data flow analysis computes the sources of values that a variable or expression can hold at points in a program, determining how those values propagate through the program, and where they are used.

But perhaps we should loop someone on the docs team into this discussion before modifying multiple places?

Reply:
Hey folks, just for clarity the CodeQL docs don't currently conform to the same (relatively strict) style standards as docs.github.com. Generally speaking this is very well written and there's an understanding that CodeQL is a complex topic and that the docs are commensurately relatively complex.

Reply:
Let's leave this as it is for now.